File: src/main/java/com/dsa/problems/scaler/hashing/count_unique_elements.java
package com.dsa.problems.scaler.hashing;

import java.util.*;

public class count_unique_elements {
  /**
   * Q. Count unique elements
   *
   * You are given an array A of N integers. Return the count of elements with frequency 1 in the given array.
   */
  public static int countUniqueElements(ArrayList<Integer> A) {
    HashMap<Integer, Integer> hm = new HashMap<>();

    for(int a : A) {
      if(hm.containsKey(a)) {
        hm.put(a, hm.get(a) + 1);
      } else {
        hm.put(a, 1);
      }
    }

    int rslt = 0;
    for(int a : A) {
      if(hm.containsKey(a) && hm.get(a) == 1) {
        rslt++;
      }
    }

    return rslt;
  }

  public static void main(String[] args) {
    System.out.println(countUniqueElements(new ArrayList<>(Arrays.asList(1,2,2,1,3,4,5,5,3))));
  }
}

File: problems/scaler/hashing/count_unique_elements.js
/**
 * Q. Count unique elements
 *
 * You are given an array A of N integers. Return the count of elements with frequncy 1 in the given array.
 */

function soln(A) {
  const mp = new Map();
  for (const i of A) {
    if (mp.has(i)) {
      mp.set(i, mp.get(i) + 1);
    } else {
      mp.set(i, 1);
    }
  }

  let rslt = 0;
  mp.forEach((v) => {
    if (v === 1) {
      rslt++;
    }
  });

  return rslt;
}

console.log(soln([3, 4, 3, 6, 6]));
